### PR TITLE
completion/aliases: eliminate use of `eval`

### LINKED ISSUE #1629: Code possibly executed unwantedly in alias completion plugin
Consider we have the `alias_completion` function from `plugins/available/alias-completion.plugin.bash` available,
and an alias like ```alias cleantmp='echo Removed `rm -v /tmp/uniquenameasdf* | wc -l` files.'```:

```
$ touch /tmp/uniquenameasdf42
$ alias_completion 
$ rm /tmp/uniquenameasdf42
rm: cannot remove '/tmp/uniquenameasdf42': No such file or directory
$
```

`eval` is dangerous!

### PR DESCRIPTION
**Pull request by gaelicWizard**
## Description
Replace misuse of `eval` with idiomatic structures that actually achieve the goal. 

This closes #1629 and might address #1995.

## Motivation and Context
Eval is the strongest code-smell of them all. It's almost universally wrong.

## How Has This Been Tested?
Works for me, all tests pass.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

**Review by NoahGorny — APPROVED** [review]
nice one @gaelicWizard !

**Comment by gaelicWizard** [comment]
@NoahGorny, I've just realized this is a breaking change due to our use of symlinks. I'll add a stub shortly. 👍

File: plugins/available/alias-completion.plugin.bash [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# shellcheck shell=bash
+# stup for renamed file
```
Comment (NoahGorny) [inline plugins/available/alias-completion.plugin.bash:2]: you probably meant stub!

Comment (gaelicWizard) [reply plugins/available/alias-completion.plugin.bash]: Fixed!

**Comment by gaelicWizard** [comment]
This might fix #2046?

**Pull request merged**